pkg/component/etcd/etcd/bootstrap.go

```diff
@@ -258,6 +290,134 @@ func (b *bootstrapper) Deploy(ctx context.Context) error {
 						Port:       metricsPort,
 						TargetPort: intstr.FromInt32(metricsPort),
 					},
+					{
+						Name:       webhookServerPortName,
+						Protocol:   corev1.ProtocolTCP,
+						Port:       webhookServerPort,
+						TargetPort: intstr.FromInt32(webhookServerPort),
+					},
+				},
+			},
+		}
+
+		opUpdateAndDelete = []admissionregistrationv1.OperationType{admissionregistrationv1.Update, admissionregistrationv1.Delete}
+		opDelete          = []admissionregistrationv1.OperationType{admissionregistrationv1.Delete}
+		clientConfig      = admissionregistrationv1.WebhookClientConfig{
+			Service: &admissionregistrationv1.ServiceReference{
+				Name:      druidServiceName,
+				Namespace: b.namespace,
+				Path:      ptr.To[string]("/webhooks/etcdcomponents"),
+				Port:      ptr.To[int32](webhookServerPort),
+			},
+			CABundle: caSecret.Data[secretsutils.DataKeyCertificateBundle],
+		}
+		validatingWebhookConfiguration = &admissionregistrationv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      druidWebhookName,
+				Namespace: b.namespace,
+				Labels:    labels(),
+			},
+			Webhooks: []admissionregistrationv1.ValidatingWebhook{
+				{
+					Name:                    "etcdcomponents.webhooks.druid.gardener.cloud",
+					ClientConfig:            clientConfig,
+					FailurePolicy:           ptr.To(admissionregistrationv1.Fail),
+					MatchPolicy:             ptr.To(admissionregistrationv1.Exact),
+					SideEffects:             ptr.To(admissionregistrationv1.SideEffectClassNone),
+					TimeoutSeconds:          ptr.To[int32](10),
+					AdmissionReviewVersions: []string{"v1", "v1beta1"},
+					ObjectSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						druidv1alpha1.LabelManagedByKey: druidv1alpha1.LabelManagedByValue,
+					}},
+					Rules: []admissionregistrationv1.RuleWithOperations{
+						{
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{""},
```

Comment:
```suggestion
								APIGroups:   []string{corev1.SchemeGroupVersion.String()},
```

similar of other occurrences

Reply:
This will set the GroupName to "v1" instead of "". When creating ClusterRole we are using:
```go
{
	APIGroups: []string{corev1.GroupName},
	Resources: []string{"events"},
	Verbs:     []string{"create", "get", "list", "watch", "patch", "update"},
},

```

Reply:
done the changes

Reply:
My mistake - using the `GroupName` constant was correct. Please follow this approach

Reply:
Done